benches/store.rs

```diff
@@ -0,0 +1,257 @@
+/// sizes are limited to that
+fn get_path_merkletree(c: &mut Criterion) {
+    let mut group = c.benchmark_group("get_path");
+
+    let random_data_size = BATCH_SIZES.into_iter().max().unwrap();
+    let random_data: Vec<RpoDigest> = (0..random_data_size)
+        .map(|_| rand_array::<Felt, 4>().into())
+        .collect();
+
+    for size in BATCH_SIZES {
+        let leaves = &random_data[..size];
+
+        let mtree_leaves: Vec<Word> = leaves.iter().map(|v| v.into()).collect();
+        let mtree = MerkleTree::new(mtree_leaves.clone()).unwrap();
+        let store = MerkleStore::new().with_merkle_tree(mtree_leaves).unwrap();
+        let depth = mtree.depth();
+        let root = mtree.root();
+
+        group.bench_function(BenchmarkId::new(format!("MerkleTree"), size), |b| {
+            b.iter_batched(
+                || rand_value::<u64>() % u64::try_from(size).unwrap(),
+                |value| black_box(mtree.get_path(NodeIndex::new(depth, value))),
+                BatchSize::SmallInput,
+            )
+        });
+
+        group.bench_function(BenchmarkId::new(format!("MerkleStore"), size), |b| {
+            b.iter_batched(
+                || rand_value::<u64>() % u64::try_from(size).unwrap(),
+                |value| black_box(store.get_path(root, NodeIndex::new(depth, value))),
+                BatchSize::SmallInput,
+            )
+        });
+    }
+}
+
+fn get_path_simplesmt(c: &mut Criterion) {
+    let mut group = c.benchmark_group("get_path");
+
+    let random_data_size = BATCH_SIZES.into_iter().max().unwrap();
+    let random_data: Vec<RpoDigest> = (0..random_data_size)
+        .map(|_| rand_array::<Felt, 4>().into())
+        .collect();
+
+    for size in BATCH_SIZES {
+        let leaves = &random_data[..size];
+
+        let smt_leaves = leaves
+            .iter()
+            .enumerate()
+            .map(|(c, v)| (c.try_into().unwrap(), v.into()))
+            .collect::<Vec<(u64, Word)>>();
+        let smt = SimpleSmt::new(63)
+            .unwrap()
+            .with_leaves(smt_leaves.clone())
+            .unwrap();
+        let store = MerkleStore::new()
+            .with_sparse_merkle_tree(smt_leaves)
+            .unwrap();
+        let depth = smt.depth();
+        let root = smt.root();
+
+        group.bench_function(BenchmarkId::new(format!("SimpleSmt"), size), |b| {
+            b.iter_batched(
+                || rand_value::<u64>() % u64::try_from(size).unwrap(),
+                |value| black_box(smt.get_path(NodeIndex::new(depth, value))),
+                BatchSize::SmallInput,
+            )
+        });
+
+        group.bench_function(BenchmarkId::new(format!("MerkleStore"), size), |b| {
+            b.iter_batched(
+                || rand_value::<u64>() % u64::try_from(size).unwrap(),
+                |value| black_box(store.get_path(root, NodeIndex::new(depth, value))),
+                BatchSize::SmallInput,
+            )
+        });
```

Comment:
As mentioned in https://github.com/0xPolygonMiden/crypto/pull/97#issuecomment-1474698263, assuming I understand it correctly, this benchmark seems to produce very counter-intuitive results. We should try to understand why.

Also, we should probably test building paths leading to empty nodes as well (similar to my other comment above).

Lastly, the ID of the benchmark conflicts with another benchmark above - maybe change it slightly here?

Reply:
changed the benchmark group names to `get_path_merkletree` and `get_path_simplesmt`